Silence a warning that fails builds with GCC 14, especially in XNNPACK.
See https://github.com/pytorch/pytorch/pull/166873

Applied more broadly as we don't care about warnings anyway.

Author: Alexander Grund (TU Dresden)

diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1056,6 +1056,12 @@ if(NOT MSVC)
     string(APPEND CMAKE_CXX_FLAGS " -Wno-psabi")
   endif()
 
+  if(CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL "14")
+    foreach(xnn_tgt IN ITEMS XNNPACK microkernels-prod microkernels-all)
+      string(APPEND CMAKE_CXX_FLAGS " -Wno-incompatible-pointer-types")
+      string(APPEND CMAKE_C_FLAGS " -Wno-incompatible-pointer-types")
+    endforeach()
+  endif()
   # Use ld.gold if available, fall back to ld.bfd (the default ld) if not
   if(USE_GOLD_LINKER)
     if(USE_DISTRIBUTED AND USE_MPI)
